libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -91,11 +104,30 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		return nil, fmt.Errorf("Can not initialize kubernetes plugin with zero matcher plugins")
 	}
 
+	processor := &kubernetesAnnotator{
+		indexers: indexers,
+		matchers: matchers,
+		cache:    newCache(config.CleanupTimeout),
+	}
+
 	client, err := kubernetes.GetKubernetesClient(config.KubeConfig)
 	if err != nil {
-		return nil, err
+		if config.InCluster {
```

Comment:
`config.InCluster` could be replaced by `kubernetes.IsInCluster(config.KubeConfig)` as used in other places. Or at least it should be consistent with the result of this call.